src/agent/coverage/src/block/windows.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::BTreeMap;
+use std::process::Command;
+use std::time::{Duration, Instant};
+
+use anyhow::Result;
+use debugger::{
+    dbghelp::SymInfo,
+    debugger::{BreakpointId, BreakpointType, DebugEventHandler, Debugger},
+    target::Module,
+};
+
+use crate::{AppCoverageBlocks, ModuleCoverageBlocks};
+
+pub fn record(cmd: Command) -> Result<AppCoverageBlocks> {
+    let mut handler = BlockCoverageHandler::new();
+
+    let (mut dbg, _child) = Debugger::init(cmd, &mut handler)?;
+    dbg.run(&mut handler)?;
+
+    Ok(handler.coverage)
+}
+
+pub struct BlockCoverageHandler {
+    bp_to_block: BTreeMap<BreakpointId, (usize, usize)>,
+    coverage: AppCoverageBlocks,
+    started: Instant,
+    max_duration: Duration,
+    timed_out: bool,
+}
+
+impl BlockCoverageHandler {
+    pub fn new() -> Self {
+        let coverage = AppCoverageBlocks::new();
+        let bp_to_block = BTreeMap::default();
+        let started = Instant::now();
+        let max_duration = Duration::from_secs(5);
+        let timed_out = false;
+
+        Self {
+            bp_to_block,
+            coverage,
+            max_duration,
+            started,
+            timed_out,
+        }
+    }
+
+    pub fn pc(&self, dbg: &mut Debugger) -> Result<(u64, Option<SymInfo>)> {
+        let rip = iced_x86::Register::RIP;
```

Comment:
Any reason to not inline this, perhaps with `use`?

Reply:
Sure, I think that'd be more idiomatic.